ssl/statem/extensions.c

```diff
@@ -1731,3 +1747,137 @@ static int final_psk(SSL_CONNECTION *s, unsigned int context, int sent)
 
     return 1;
 }
+
+static int tls_init_compress_certificate(SSL_CONNECTION *sc, unsigned int context)
+{
+    sc->ext.compress_certificate_from_peer = TLSEXT_comp_cert_none;
+    return 1;
+}
+
+/* The order these are put into the packet imply a preference order: [brotli, zlib, zstd] */
+static EXT_RETURN tls_construct_compress_certificate(SSL_CONNECTION *sc, WPACKET *pkt,
+                                                     unsigned int context,
+                                                     X509 *x, size_t chainidx)
+{
+#if !defined(OPENSSL_NO_COMP) && (defined(ZLIB) || !defined(OPENSSL_NO_BROTLI) || !defined(OPENSSL_NO_ZSTD))
```

Comment:
It seems like we should have a simplified macro for this combination. We repeat it quite a bit.

Reply:
Agreed, good point.

Reply:
Surprisingly, on 7 times...